e2e/cypress/integration/messaging/emoji_to_markdown_spec.js

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+function createMessages(message) {
+    const aliasLineMessage = 'newLineMessage';
+    const aliasLineMessageReplyInRHS = 'newLineMessageReplyInRHS';
+
+    cy.postMessage(message);
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).as(aliasLineMessage);
+        cy.clickPostCommentIcon(postId);
+    });
+
+    cy.postMessageReplyInRHS(message + '{enter}');
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).as(aliasLineMessageReplyInRHS);
+    });
+
+    return [aliasLineMessage, aliasLineMessageReplyInRHS];
+}
+
+function createAndVerifyMessage(message, isCode) {
+    if (isCode) {
+        const aliases = createMessages(message);
+
+        aliases.forEach((alias) => {
+            cy.get('@' + alias).
+                children().should('have.class', 'post-code').
+                children('code').should('be.visible').contains(message.trim());
+        });
+    } else {
+        const aliases = createMessages(message);
```

Comment:
nit: Please move this up, outside the if-else block so it's written ones only.

Reply:
Right, moved it.